File: CommercialControllerC#/CommercialController.cs
using System;
using System.Collections.Generic;
using System.Threading;

//-------------------------------------"    Initializing the Main System   "-------------------------------------

public class ElevatorController
{
    public int numberFloor;
    public int elevatorPerColumn;
    public int numberColumn;
    public string userDirection;
    public Battery battery;

    public ElevatorController(int numberFloor, int numberColumn, int elevatorPerColumn, string userDirection)
    {
        this.numberFloor = numberFloor;
        this.numberColumn = numberColumn;
        this.elevatorPerColumn = elevatorPerColumn;
        this.userDirection = userDirection;
        this.battery = new Battery(this.numberColumn);


    }

//-------------------------------------"    User Requests Elevator to Return to Lobby   "-------------------------------------

    public Elevator RequestElevatorReturning(int FloorNumber, int RequestedFloor)
    {
        Thread.Sleep(200);
        Console.WriteLine("Call Button Activated");
        Console.WriteLine("====================================================");
        Console.WriteLine("Request elevator to floor : " + FloorNumber);
        Console.WriteLine("====================================================");
        Thread.Sleep(200);
        var column = battery.SelectAppropriateColumn(FloorNumber);
        userDirection = "down";
        var elevator = column.selectOptimalElevatorReturning(FloorNumber, userDirection);
        if (elevator.elevatorFloor > FloorNumber)
        {
            elevator.addtoFloorQueue(FloorNumber, column.columnNumber);
            Console.WriteLine("Lobby Button Activated");
            Console.WriteLine("====================================================");
            elevator.addtoFloorQueue(RequestedFloor, column.columnNumber);
        }

        else if (elevator.elevatorFloor < FloorNumber)
        {
           
            elevator.addtoFloorQueue(FloorNumber, column.columnNumber);
            Console.WriteLine("Lobby Button Activated");
            Console.WriteLine("====================================================");
            elevator.addtoFloorQueue(RequestedFloor, column.columnNumber);
        }

        return elevator;
        
    }

//-------------------------------------"    User Requests Elevator From Lobby   "-------------------------------------
    public Elevator RequestElevator(int RequestedFloor)
    {
        Thread.Sleep(200);
        Console.WriteLine("Requested floor : " + RequestedFloor);
        Thread.Sleep(200);
        Console.WriteLine("Call Button Activated");


        Column column = battery.SelectAppropriateColumn(RequestedFloor);
        userDirection = "up";
        var FloorNumber = 1;
        var elevator = column.selectOptimalElevator(RequestedFloor, FloorNumber, userDirection);

        elevator.addtoFloorQueue(FloorNumber, column.columnNumber);
        elevator.addtoFloorQueue(RequestedFloor, column.columnNumber);



        return elevator;
    }


//-------------------------------------"    Initializing Battery    "-------------------------------------

    public class Battery
    {
        public string batteryStatus;
        public int numberColumn;
        public List<Column> columnList;


        public Battery(int numberColumn)
        {
            this.numberColumn = numberColumn;
            this.batteryStatus = "Activated";
            columnList = new List<Column>();



            char cols = 'A';
            for (int i = 0; i < this.numberColumn; i++, cols++)
            {

                Column column = new Column(cols, 66, 5);

                column.columnNumber = cols;
                columnList.Add(column);

            }
        }

//-------------------------------------"    Selecting the Appropriate Column    "-------------------------------------

        public Column SelectAppropriateColumn(int RequestedFloor)
        {
            Column appropriateColumn = null;
            foreach (Column column in columnList)
            {
                if (RequestedFloor > -6 && RequestedFloor <= -1 || RequestedFloor == 1)
                {
                    appropriateColumn = columnList[0];
                }
                else if (RequestedFloor >= 1 && RequestedFloor <= 20 || RequestedFloor == 1)
                {

                    appropriateColumn = columnList[1];


                }
                else if (RequestedFloor >= 21 && RequestedFloor <= 40 || RequestedFloor == 1)
                {
                    appropriateColumn = columnList[2];


                }
                else if (RequestedFloor >= 41 && RequestedFloor <= 60 || RequestedFloor == 1)
                {
                    appropriateColumn = columnList[3];


                }

            }
            return appropriateColumn;
        }
    }

//-------------------------------------"    Initializing Column    "-------------------------------------

    public class Column
    {
        public char columnNumber;
        public int numberFloor;
        public int elevatorPerColumn;
        public List<Elevator> elevatorList;
        public List<int> lobbyCallButtonList;


        public Column(char columnNumber, int numberFloor, int elevatorPerColumn)
        {
            this.columnNumber = columnNumber;
            this.numberFloor = numberFloor;
            this.elevatorPerColumn = elevatorPerColumn;
            elevatorList = new List<Elevator>();
            lobbyCallButtonList = new List<int>();
            for (int i = 0; i < this.elevatorPerColumn; i++)
            {
                Elevator elevator = new Elevator(i, "idle", 1, "up");
                elevatorList.Add(elevator);
            }
        }

//-------------------------------------"    Selecting the Optimal Elevator  "-------------------------------------

        public Elevator selectOptimalElevatorReturning(int RequestedFloor, string userDirection)
        {
            var smallestDistance = 999;
            var optimalElevator = 0;

            for (var i = 0; i < this.elevatorList.Count; i++)
            {
                var distance = elevatorList[i].elevatorFloor - RequestedFloor;

                if (distance > 0 && distance < smallestDistance)
                {
                    smallestDistance = distance;
                    optimalElevator = i;
                }
            }
            return elevatorList[optimalElevator];
        }

        public Elevator selectOptimalElevator(int RequestedFloor, int FloorNumber, string userDirection)
        {

            foreach (var elevator in elevatorList)
                if (elevator.status == "idle")
                {
                    return elevator;
                }
                else if (elevator.status == "stopped" || FloorNumber == RequestedFloor)
                {
                    return elevator;
                }
            var optimalElevator = 0;
            var smallestDistance = 1000;
            for (var i = 0; i < this.elevatorList.Count; i++)
            {
                var distance = Math.Abs(elevatorList[i].elevatorFloor - elevatorList[i].floorQueue[0]) + Math.Abs(elevatorList[i].floorQueue[0] - 1);
                if (smallestDistance >= distance)
                {
                    smallestDistance = distance;
                    optimalElevator = i;
                }
            }
            return elevatorList[optimalElevator];
        }
    }
    
//-------------------------------------"    Initializing Elevator  "-------------------------------------

    public class Elevator
    {
        public int elevatorNumber;
        public string status;
        public int elevatorFloor;
        public string elevatorDirection;
        public bool doorClearance;
        public int FloorDisplay;
        public List<int> floorQueue;

        public Elevator(int elevatorNumber, string status, int elevatorFloor, string elevatorDirection)
        {
            this.elevatorNumber = elevatorNumber;
            this.status = status;
            this.elevatorFloor = elevatorFloor;
            this.elevatorDirection = elevatorDirection;
            this.FloorDisplay = elevatorFloor;
            this.doorClearance = true;
            this.floorQueue = new List<int>();
        }

//-------------------------------------"    Pushing and Sorting the Requested Elevator in the Floor Queue  "-------------------------------------

        public void addtoFloorQueue(int RequestedFloor, char columnNumber)
        {
            floorQueue.Add(RequestedFloor);
            if (RequestedFloor > elevatorFloor)
            {
                floorQueue.Sort((a, b) => a.CompareTo(b));
            }
            else if (RequestedFloor < elevatorFloor)
            {
                floorQueue.Sort((a, b) => -1 * a.CompareTo(b));

            }

            MoveElevator(RequestedFloor, columnNumber);
        }

//-------------------------------------"    Moving the Elevator  "-------------------------------------

        public void MoveElevator(int RequestedFloor, char columnNumber)
        {
            if (RequestedFloor == elevatorFloor)
            {
                OpenDoors();
                this.status = "moving";

                this.floorQueue.Remove(0);
            }
            else if (RequestedFloor < this.elevatorFloor)
            {
                status = "moving";
                Console.WriteLine("Column: " + columnNumber + " Elevator: " + this.elevatorNumber + " is " + status);
                Console.WriteLine("====================================================");
                this.elevatorDirection = "down";
                MoveDown(RequestedFloor, columnNumber);
                this.status = "stopped";
                Console.WriteLine("Column: " + columnNumber + " Elevator: " + this.elevatorNumber + " has " + status);
                Console.WriteLine("====================================================");
                Console.WriteLine("Button Disactivated");
                
                this.OpenDoors();
                this.floorQueue.Remove(0);
            }
            else if (RequestedFloor > this.elevatorFloor)
            {
                Thread.Sleep(700);
                this.status = "moving";
                Console.WriteLine("Column: " + columnNumber + " Elevator: " + this.elevatorNumber + " is " + status);
                Console.WriteLine("====================================================");
                this.elevatorDirection = "up";
                this.MoveUp(RequestedFloor, columnNumber);
                this.status = "stopped";
                Console.WriteLine("====================================================");
                Console.WriteLine("Column: " + columnNumber + " Elevator: " + this.elevatorNumber + " has " + status);
                Console.WriteLine("====================================================");
                Console.WriteLine("Button Disactivated");

                this.OpenDoors();

                this.floorQueue.Remove(0);
            }

        }

//-------------------------------------"    Moving Up  "-------------------------------------

        public void MoveUp(int RequestedFloor, char columnNumber)
        {
            Console.WriteLine("Column: " + columnNumber + " Elevator: " + elevatorNumber + "  Current Floor: " + this.elevatorFloor);
            Thread.Sleep(700);
            Console.WriteLine("====================================================");
            while (this.elevatorFloor != RequestedFloor)
            {
                this.elevatorFloor += 1;
                Console.WriteLine("Column: " + columnNumber + " Elevator: " + elevatorNumber + "  Floor: " + this.elevatorFloor);

                Thread.Sleep(250);
            }
          
        }

//-------------------------------------"    Moving Down  "-------------------------------------

        public void MoveDown(int RequestedFloor, char columnNumber)
        {
            Console.WriteLine("Column: " + columnNumber + " Elevator: " + elevatorNumber + "  Current Floor: " + this.elevatorFloor);
            Thread.Sleep(700);
            Console.WriteLine("====================================================");

            while (this.elevatorFloor != RequestedFloor)
            {
                this.elevatorFloor -= 1;
                Console.WriteLine("Column: " + columnNumber + " Elevator: " + elevatorNumber + "  Floor: " + this.elevatorFloor);

                Thread.Sleep(250);
            }
            Console.WriteLine("====================================================");

        }

//-------------------------------------"    Doors Status    "-------------------------------------

        public void OpenDoors()
        {
            Thread.Sleep(700);

            Console.WriteLine("====================================================");
            Console.WriteLine("Doors Opening");

            Thread.Sleep(700);
            Console.WriteLine("Doors are Opened");
            Thread.Sleep(700);

            this.CloseDoors();
        }
        public void CloseDoors()
        {
            if (doorClearance == true)
            {
                Console.WriteLine("Doors Closing");
                Thread.Sleep(700);
                Console.WriteLine("Doors are Closed");
                Thread.Sleep(700);


                Console.WriteLine("====================================================");
            }
            else if (doorClearance == false)
            {
                OpenDoors();
            }
        }
    }
}

//-------------------------------------"   Scenario Testing   "-------------------------------------

    public class CommercialControllerTesting
    {
        public static void Main(string[] args)
        {
                
//-------------------------------------"   Scenario 1   "-------------------------------------  

        void Scenario1() 
        {
            ElevatorController controller = new ElevatorController(66, 4, 5, "up");

            controller.battery.columnList[1].elevatorList[0].elevatorFloor = 20;
            controller.battery.columnList[1].elevatorList[0].elevatorDirection = "down";
            controller.battery.columnList[1].elevatorList[0].status = "moving";
            controller.battery.columnList[1].elevatorList[0].floorQueue.Add(5);


            controller.battery.columnList[1].elevatorList[1].elevatorFloor = 3;
            controller.battery.columnList[1].elevatorList[1].elevatorDirection = "up";
            controller.battery.columnList[1].elevatorList[1].status = "moving";
            controller.battery.columnList[1].elevatorList[1].floorQueue.Add(15);


            controller.battery.columnList[1].elevatorList[2].elevatorFloor = 13;
            controller.battery.columnList[1].elevatorList[2].elevatorDirection = "down";
            controller.battery.columnList[1].elevatorList[2].status = "moving";
            controller.battery.columnList[1].elevatorList[2].floorQueue.Add(1);


            controller.battery.columnList[1].elevatorList[3].elevatorFloor = 15;
            controller.battery.columnList[1].elevatorList[3].elevatorDirection = "down";
            controller.battery.columnList[1].elevatorList[3].status = "moving";
            controller.battery.columnList[1].elevatorList[3].floorQueue.Add(2);
           


            controller.battery.columnList[1].elevatorList[4].elevatorFloor = 6;
            controller.battery.columnList[1].elevatorList[4].elevatorDirection = "down";
            controller.battery.columnList[1].elevatorList[4].status = "moving";
            controller.battery.columnList[1].elevatorList[4].floorQueue.Add(1);



            controller.RequestElevator(20);
        }

//-------------------------------------"   Scenario 2   "-------------------------------------

        void Scenario2() 
        {
            ElevatorController controller = new ElevatorController(66, 4, 5, "up");

            controller.battery.columnList[2].elevatorList[0].elevatorFloor = 1;
            controller.battery.columnList[2].elevatorList[0].elevatorDirection = "up";
            controller.battery.columnList[2].elevatorList[0].status = "stopped";
            controller.battery.columnList[2].elevatorList[0].floorQueue.Add(21);


            controller.battery.columnList[2].elevatorList[1].elevatorFloor = 23;
            controller.battery.columnList[2].elevatorList[1].elevatorDirection = "up";
            controller.battery.columnList[2].elevatorList[1].status = "moving";
            controller.battery.columnList[2].elevatorList[1].floorQueue.Add(28);


            controller.battery.columnList[2].elevatorList[2].elevatorFloor = 33;
            controller.battery.columnList[2].elevatorList[2].elevatorDirection = "down";
            controller.battery.columnList[2].elevatorList[2].status = "moving";
            controller.battery.columnList[2].elevatorList[2].floorQueue.Add(1);


            controller.battery.columnList[2].elevatorList[3].elevatorFloor = 40;
            controller.battery.columnList[2].elevatorList[3].elevatorDirection = "down";
            controller.battery.columnList[2].elevatorList[3].status = "moving";
            controller.battery.columnList[2].elevatorList[3].floorQueue.Add(24);


            controller.battery.columnList[2].elevatorList[4].elevatorFloor = 39;
            controller.battery.columnList[2].elevatorList[4].elevatorDirection = "down";
            controller.battery.columnList[2].elevatorList[4].status = "moving";
            controller.battery.columnList[2].elevatorList[4].floorQueue.Add(1);

            controller.RequestElevator(36);
        }

//-------------------------------------"   Scenario 3   "------------------------------------- 

        void Scenario3() 
        {
            ElevatorController controller = new ElevatorController(66, 4, 5, "down");   

            controller.battery.columnList[3].elevatorList[0].elevatorFloor = 58;
            controller.battery.columnList[3].elevatorList[0].elevatorDirection = "down";
            controller.battery.columnList[3].elevatorList[0].status = "moving";
            controller.battery.columnList[3].elevatorList[0].floorQueue.Add(1);


            controller.battery.columnList[3].elevatorList[1].elevatorFloor = 50;
            controller.battery.columnList[3].elevatorList[1].elevatorDirection = "up";
            controller.battery.columnList[3].elevatorList[1].status = "moving";
            controller.battery.columnList[3].elevatorList[1].floorQueue.Add(60);


            controller.battery.columnList[3].elevatorList[2].elevatorFloor = 46;
            controller.battery.columnList[3].elevatorList[2].elevatorDirection = "up";
            controller.battery.columnList[3].elevatorList[2].status = "moving";
            controller.battery.columnList[3].elevatorList[2].floorQueue.Add(58);


            controller.battery.columnList[3].elevatorList[3].elevatorFloor = 1;
            controller.battery.columnList[3].elevatorList[3].elevatorDirection = "up";
            controller.battery.columnList[3].elevatorList[3].status = "moving";
            controller.battery.columnList[3].elevatorList[3].floorQueue.Add(54);


            controller.battery.columnList[3].elevatorList[4].elevatorFloor = 60;
            controller.battery.columnList[3].elevatorList[4].elevatorDirection = "down";
            controller.battery.columnList[3].elevatorList[4].status = "moving";
            controller.battery.columnList[3].elevatorList[4].floorQueue.Add(1);



            controller.RequestElevatorReturning(54, 1);
        }
    
        //Scenario1();
        //Scenario2();
        Scenario3();
        }
    }

File: CommercialControllerGo/CommercialController.go
package main

import (
	"fmt"
	"sort"
	"time"
)

//-------------------------------------"    Main System Structure   "-------------------------------------

type ElevatorController struct {
	numberBattery int
	battery       []Battery
	numberColumn   int
	userDirection  string
}

//-------------------------------------"    Battery Structure   "-------------------------------------

type Battery struct {
	numberColumn int
	columnList   []Column
}

//-------------------------------------"    Column Structure   "-------------------------------------

type Column struct {
	columnNumber               int
	elevatorPerColumn int
	elevatorList           []Elevator
}

//-------------------------------------"    Elevator Structure   "-------------------------------------

type Elevator struct {
	elevatorNumber        int
	elevatorFloor  int
	floorQueue        []int
	elevatorStatus    string
	elevatorDirection string
	doorClearance       bool
	column             int
	columnNumber               int
}

//-------------------------------------"    Creating the Battery for the Main System  "-------------------------------------

func NewController(numberBattery int) *ElevatorController {
	controller := new(ElevatorController)
	controller.numberBattery = 1
	for i := 0; i < numberBattery; i++ {
		battery := NewBattery(i)
		controller.battery = append(controller.battery, *battery)
	}
	return controller
}

//-------------------------------------"    Creating the Column in the Battery  "-------------------------------------

func NewBattery(numberColumn int) *Battery {
	battery := new(Battery)
	battery.numberColumn = 4
	for i := 0; i < battery.numberColumn; i++ {
		column := NewColumn(i)
		battery.columnList = append(battery.columnList, *column)
	}
	return battery
}

//-------------------------------------"    Creating the Elevator in the Column  "-------------------------------------

func NewColumn(elevatorPerColumn int) *Column {
	column := new(Column)
	column.elevatorPerColumn = 5
	for i := 0; i < column.elevatorPerColumn; i++ {
		elevator := NewElevator()
		column.elevatorList = append(column.elevatorList, *elevator)
	}
	return column
}

//-------------------------------------"    Creating the Elevator   "-------------------------------------

func NewElevator() *Elevator {
	elevator := new(Elevator)
	elevator.elevatorFloor = 1
	elevator.floorQueue = []int{}
	elevator.elevatorStatus = "idle"
	elevator.elevatorDirection = "up"
	elevator.doorClearance= true
	return elevator
}

//-------------------------------------"    User Requests Elevator to Return to Lobby   "-------------------------------------

func (controller *ElevatorController) RequestElevatorReturning(FloorNumber, RequestedFloor int) Elevator {
	fmt.Println("Elevator Requested on Floor : ", FloorNumber)
	var column = controller.battery[0].SelectAppropriateColumn(FloorNumber)
	controller.userDirection = "down"
	var elevator = column.SelectOptimalElevator(RequestedFloor, controller.userDirection)
	elevator.addtoFloorQueue(FloorNumber)
	elevator.addtoFloorQueue(RequestedFloor)
	return elevator
}

//-------------------------------------"    User Requests Elevator from Lobby   "-------------------------------------

func (controller *ElevatorController) RequestElevator(RequestedFloor int) Elevator {
	fmt.Println("Request elevator to floor : ", RequestedFloor)
	column := controller.battery[0].SelectAppropriateColumn(RequestedFloor)
	controller.userDirection = "up"
	var elevator = column.SelectOptimalElevator(RequestedFloor, controller.userDirection)
	var FloorNumber = 1
	elevator.addtoFloorQueue(FloorNumber)
	elevator.addtoFloorQueue(RequestedFloor)
	return elevator
}

//-------------------------------------"    Selecting the Appropriate Column  "-------------------------------------

func (b *Battery) SelectAppropriateColumn(RequestedFloor int) Column { // not sure about *
	if RequestedFloor > -6 && RequestedFloor <= 1 {
		return b.columnList[0]
	} else if RequestedFloor > 1 && RequestedFloor <= 20 {
		return b.columnList[1]
	} else if RequestedFloor > 21 && RequestedFloor <= 40 {
		return b.columnList[2]
	} else if RequestedFloor > 41 && RequestedFloor <= 60 {
		return b.columnList[3]
	}
	return b.columnList[3]
}

//-------------------------------------"    Selecting the Optimal Elevator  "-------------------------------------

func (c *Column) SelectOptimalElevator(RequestedFloor int, userDirection string) Elevator {
	var optimalElevator = c.elevatorList[0]
	for _, e := range c.elevatorList {
		if RequestedFloor < e.elevatorFloor && e.elevatorDirection == "down" && userDirection == "down" {
			optimalElevator = e
		} else if e.elevatorStatus == "idle" {
			optimalElevator = e
		} else if e.elevatorDirection != userDirection && e.elevatorStatus == "moving" || e.elevatorStatus == "stopped" {
			optimalElevator = e
		} else if e.elevatorDirection == userDirection && e.elevatorStatus == "moving" || e.elevatorStatus == "stopped" {
			optimalElevator = e
		}
	}
	return optimalElevator
}

//-------------------------------------"    Pushing and Sorting the Requested Elevator in the Floor Queue  "-------------------------------------

func (e *Elevator) addtoFloorQueue(RequestedFloor int) {
	e.floorQueue = append(e.floorQueue, RequestedFloor)
	if RequestedFloor > e.elevatorFloor {

		sort.Ints(e.floorQueue)
	} else if RequestedFloor < e.elevatorFloor {

		sort.Sort(sort.Reverse(sort.IntSlice(e.floorQueue)))
	}
	e.MoveElevator(RequestedFloor)
}

//-------------------------------------"    Moving the Elevator   "-------------------------------------

func (e *Elevator) MoveElevator(RequestedFloor int) {
	if RequestedFloor == e.elevatorFloor {
		e.OpenDoors()
	} else if RequestedFloor > e.elevatorFloor {
		e.elevatorStatus = "moving"
		e.moveUp(RequestedFloor)
		e.elevatorStatus = "stopped"
		e.OpenDoors()
		e.elevatorStatus = "moving"
	} else if RequestedFloor < e.elevatorFloor {
		e.elevatorStatus = "moving"
		e.moveDown(RequestedFloor)
		e.elevatorStatus = "stopped"
		e.OpenDoors()
		e.elevatorStatus = "moving"
	}
}

//-------------------------------------"    Moving Up  "-------------------------------------

func (e *Elevator) moveUp(RequestedFloor int) {
	fmt.Println("Column : ", e.columnNumber, " Elevator : #", e.elevatorNumber, " Current Floor :", e.elevatorFloor)
	for RequestedFloor > e.elevatorFloor {
		e.elevatorFloor += 1
		if RequestedFloor == e.elevatorFloor {
			time.Sleep(1 * time.Second)
			fmt.Println("======================================")
			fmt.Println("Column : ", e.columnNumber, " Elevator : #", e.elevatorNumber, " Arrived at Requested Floor : ", e.elevatorFloor)
		}
		fmt.Println("Column : ", e.columnNumber, " Elevator : #", e.elevatorNumber, " Floor : ", e.elevatorFloor)
	}
}

//-------------------------------------"    Moving Down   "-------------------------------------

func (e *Elevator) moveDown(RequestedFloor int) {
	fmt.Println("Column : ", e.columnNumber, " Elevator : #", e.elevatorNumber, " Current Floor :", e.elevatorFloor)
	for RequestedFloor < e.elevatorFloor {
		e.elevatorFloor -= 1
		if RequestedFloor == e.elevatorFloor {
			time.Sleep(1 * time.Second)
			fmt.Println("======================================")
			fmt.Println("Column : ", e.columnNumber, " Elevator : #", e.elevatorNumber, " Arrived at Requested Floor : ", e.elevatorFloor)
		}
		fmt.Println("Column : ", e.columnNumber, " Elevator : #", e.elevatorNumber, " Floor : ", e.elevatorFloor)
	}
}

//-------------------------------------"    Doors Status   "-------------------------------------

func (e *Elevator) OpenDoors() {
	fmt.Println("======================================")
	fmt.Println("Doors Opening")
	fmt.Println("Doors are Open")
	e.CloseDoors()
}
func (e *Elevator) CloseDoors() {
	if e.doorClearance== true {
		fmt.Println("Doors Closing")
		fmt.Println("Doors are Close")
		fmt.Println("======================================")
	} else if e.doorClearance{
		e.OpenDoors()
	}
}

//-------------------------------------"    Scenario 1   "-------------------------------------

func (controller *ElevatorController) TestScenario1() {


	controller.battery[0].columnList[1].elevatorList[0].elevatorFloor = 20
	controller.battery[0].columnList[1].elevatorList[0].elevatorDirection = "down"
	controller.battery[0].columnList[1].elevatorList[0].elevatorStatus = "moving"
	controller.battery[0].columnList[1].elevatorList[0].floorQueue = append(controller.battery[0].columnList[1].elevatorList[0].floorQueue, 5)

	controller.battery[0].columnList[1].elevatorList[1].elevatorFloor = 3
	controller.battery[0].columnList[1].elevatorList[1].elevatorDirection = "up"
	controller.battery[0].columnList[1].elevatorList[1].elevatorStatus = "moving"
	controller.battery[0].columnList[1].elevatorList[1].floorQueue = append(controller.battery[0].columnList[1].elevatorList[1].floorQueue, 15)

	controller.battery[0].columnList[1].elevatorList[2].elevatorFloor = 13
	controller.battery[0].columnList[1].elevatorList[2].elevatorDirection = "down"
	controller.battery[0].columnList[1].elevatorList[2].elevatorStatus = "moving"
	controller.battery[0].columnList[1].elevatorList[2].floorQueue = append(controller.battery[0].columnList[1].elevatorList[2].floorQueue, 1)

	controller.battery[0].columnList[1].elevatorList[3].elevatorFloor = 15
	controller.battery[0].columnList[1].elevatorList[3].elevatorDirection = "down"
	controller.battery[0].columnList[1].elevatorList[3].elevatorStatus = "moving"
	controller.battery[0].columnList[1].elevatorList[3].floorQueue = append(controller.battery[0].columnList[1].elevatorList[3].floorQueue, 2)

	controller.battery[0].columnList[1].elevatorList[4].elevatorFloor = 6
	controller.battery[0].columnList[1].elevatorList[4].elevatorDirection = "down"
	controller.battery[0].columnList[1].elevatorList[4].elevatorStatus = "moving"
	controller.battery[0].columnList[1].elevatorList[4].floorQueue = append(controller.battery[0].columnList[1].elevatorList[4].floorQueue, 1)
	
	controller.RequestElevator(20)
}

//-------------------------------------"    Scenario 2   "-------------------------------------

func (controller *ElevatorController) TestScenario2() {


	controller.battery[0].columnList[2].elevatorList[0].elevatorFloor = 1
	controller.battery[0].columnList[2].elevatorList[0].elevatorDirection = "up"
	controller.battery[0].columnList[2].elevatorList[0].elevatorStatus = "stopped"
	controller.battery[0].columnList[2].elevatorList[0].floorQueue = append(controller.battery[0].columnList[2].elevatorList[0].floorQueue, 21)

	controller.battery[0].columnList[2].elevatorList[1].elevatorFloor = 23
	controller.battery[0].columnList[2].elevatorList[1].elevatorDirection = "up"
	controller.battery[0].columnList[2].elevatorList[1].elevatorStatus = "moving"
	controller.battery[0].columnList[2].elevatorList[1].floorQueue = append(controller.battery[0].columnList[2].elevatorList[1].floorQueue, 28)

	controller.battery[0].columnList[2].elevatorList[2].elevatorFloor = 33
	controller.battery[0].columnList[2].elevatorList[2].elevatorDirection = "down"
	controller.battery[0].columnList[2].elevatorList[2].elevatorStatus = "moving"
	controller.battery[0].columnList[2].elevatorList[2].floorQueue = append(controller.battery[0].columnList[2].elevatorList[2].floorQueue, 1)

	controller.battery[0].columnList[2].elevatorList[3].elevatorFloor = 40
	controller.battery[0].columnList[2].elevatorList[3].elevatorDirection = "down"
	controller.battery[0].columnList[2].elevatorList[3].elevatorStatus = "moving"
	controller.battery[0].columnList[2].elevatorList[3].floorQueue = append(controller.battery[0].columnList[2].elevatorList[3].floorQueue, 24)

	controller.battery[0].columnList[2].elevatorList[4].elevatorFloor = 39
	controller.battery[0].columnList[2].elevatorList[4].elevatorDirection = "down"
	controller.battery[0].columnList[2].elevatorList[4].elevatorStatus = "moving"
	controller.battery[0].columnList[2].elevatorList[4].floorQueue = append(controller.battery[0].columnList[2].elevatorList[4].floorQueue, 1)
	
	controller.RequestElevator(36)
}

//-------------------------------------"    Scenario 3   "-------------------------------------

func (controller *ElevatorController) TestScenario3() {


	controller.battery[0].columnList[3].elevatorList[0].elevatorFloor = 58
	controller.battery[0].columnList[3].elevatorList[0].elevatorDirection = "down"
	controller.battery[0].columnList[3].elevatorList[0].elevatorStatus = "moving"
	controller.battery[0].columnList[3].elevatorList[0].floorQueue = append(controller.battery[0].columnList[3].elevatorList[0].floorQueue, 1)

	controller.battery[0].columnList[3].elevatorList[1].elevatorFloor = 50
	controller.battery[0].columnList[3].elevatorList[1].elevatorDirection = "up"
	controller.battery[0].columnList[3].elevatorList[1].elevatorStatus = "moving"
	controller.battery[0].columnList[3].elevatorList[1].floorQueue = append(controller.battery[0].columnList[3].elevatorList[1].floorQueue, 60)

	controller.battery[0].columnList[3].elevatorList[2].elevatorFloor = 46
	controller.battery[0].columnList[3].elevatorList[2].elevatorDirection = "up"
	controller.battery[0].columnList[3].elevatorList[2].elevatorStatus = "moving"
	controller.battery[0].columnList[3].elevatorList[2].floorQueue = append(controller.battery[0].columnList[3].elevatorList[2].floorQueue, 58)

	controller.battery[0].columnList[3].elevatorList[3].elevatorFloor = 1
	controller.battery[0].columnList[3].elevatorList[3].elevatorDirection = "up"
	controller.battery[0].columnList[3].elevatorList[3].elevatorStatus = "moving"
	controller.battery[0].columnList[3].elevatorList[3].floorQueue = append(controller.battery[0].columnList[3].elevatorList[3].floorQueue, 54)

	controller.battery[0].columnList[3].elevatorList[4].elevatorFloor = 60
	controller.battery[0].columnList[3].elevatorList[4].elevatorDirection = "down"
	controller.battery[0].columnList[3].elevatorList[4].elevatorStatus = "moving"
	controller.battery[0].columnList[3].elevatorList[4].floorQueue = append(controller.battery[0].columnList[3].elevatorList[4].floorQueue, 1)
	
	controller.RequestElevator(36)
}

//-------------------------------------"    Scenario 4   "-------------------------------------

func (controller *ElevatorController) TestScenario4() {


	controller.battery[0].columnList[0].elevatorList[0].elevatorFloor = -4
	controller.battery[0].columnList[0].elevatorList[0].elevatorStatus = "idle"

	controller.battery[0].columnList[0].elevatorList[1].elevatorFloor = 1
	controller.battery[0].columnList[0].elevatorList[1].elevatorStatus = "idle"

	controller.battery[0].columnList[0].elevatorList[2].elevatorFloor = -3
	controller.battery[0].columnList[0].elevatorList[2].elevatorDirection = "down"
	controller.battery[0].columnList[0].elevatorList[2].elevatorStatus = "moving"
	controller.battery[0].columnList[0].elevatorList[2].floorQueue = append(controller.battery[0].columnList[0].elevatorList[2].floorQueue, -5)

	controller.battery[0].columnList[0].elevatorList[3].elevatorFloor = -6
	controller.battery[0].columnList[0].elevatorList[3].elevatorDirection = "up"
	controller.battery[0].columnList[0].elevatorList[3].elevatorStatus = "moving"
	controller.battery[0].columnList[0].elevatorList[3].floorQueue = append(controller.battery[0].columnList[0].elevatorList[3].floorQueue, 1)

	controller.battery[0].columnList[0].elevatorList[4].elevatorFloor = -1
	controller.battery[0].columnList[0].elevatorList[4].elevatorDirection = "down"
	controller.battery[0].columnList[0].elevatorList[4].elevatorStatus = "moving"
	controller.battery[0].columnList[0].elevatorList[4].floorQueue = append(controller.battery[0].columnList[0].elevatorList[4].floorQueue, -6)
	
	controller.RequestElevator(36)
}


func main() {
	controller := NewController(1)
	controller.TestScenario1()
	//controller.TestScenario2()
	//controller.TestScenario3()
	//controller.TestScenario4()
}